client/state/utils.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * External dependencies
+ */
+import tv4 from 'tv4';
+
+/**
+ * Internal dependencies
+ */
+import warn from 'lib/warn';
+
+/**
+ * Module variables
+ */
+
+export function isValidStateWithSchema( state, schema, checkForCycles = false, banUnknownProperties = false ) {
+	const result = tv4.validateResult( state, schema, checkForCycles, banUnknownProperties );
```

Comment:
how does banUnknown defaulting to false play with `additionalProperties: false` in the actual schema? who wins?